File: cercle/Cercle.cpp
#include "Cercle.h"
//Cercle.cpp
#include "Point.h"
#include <iostream>
using namespace std;
Cercle::Cercle(Point & p0,int r):centre(p0),rayon(r){
cout << "Constructeur de Cercle avec 2 args "<< endl;
}
Cercle::Cercle(int x0,int y0,int r):centre(x0,y0),rayon(r) {
cout << "Constructeur de Cercle avec 3 args "<< endl;
}
Cercle::~Cercle() {}
void Cercle::deplacer(int dx, int dy){
this->centre.deplacer(dx,dy);
}
void Cercle::afficher(){
cout << "Le rayon = "<< rayon<< " et le centre est ";
this->centre.afficher();
}

File: Section_Java/Point/Cercle.java
public class Cercle {
    private double r;
    private Point p;
    private final double PI=3.14;
    public Cercle(double r, int x, int y){
        this.r=r;
        p=new Point(x,y);
    }
    public double Surface(){
        return this.PI*this.r*this.r;
    }
    public void affichage(){
        System.out.println("Le cercle de rayon :"+r);
        System.out.println("De centre:");
        p.afficher();
    }
    public void imprimer(){
        System.out.println("La surface du cercle est:"+Surface());

    }
}
